railties/lib/rails/generators/rails/app/templates/config/initializers/debugger.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+return unless Rails.env.local?
+
+begin
+  if Rails.env.test? ||
+    defined?(Rails::Console) ||
+    (defined?(Rake.application.top_level_tasks) && Rake.application.top_level_tasks.any?)
+
+    # Simply require `debug`, and use standard `binding.break`
+    require 'debug'
+  elsif ENV['RUBY_DEBUG_PORT'] && (defined?(Rails::Server) || ENV['PWD'].include?('.puma-dev'))
+    # When ENV['RUBY_DEBUG_PORT'] is set, debug/open_nonstop will start a TCP/IP debugger server
+    # on the provided port, which your editor can connect to.
+    #
+    # https://github.com/ruby/debug/blob/master/lib/debug/open_nonstop.rb
+    require 'debug'
+    require 'debug/open_nonstop'
```

Comment:
Wouldn't be easier to add this to debug gem to make this required based on environment variable or isn't it even possible already to call this in `Gemfile` with `require`? This whole setup seems to cryptic to be default initializer, since all it does it requires `debug` or `debug/open_nonstop`. Alternativelly you can try to create `debug-rails` gem to connect smoothly debug and Rails itself.

Reply:
Agreed